File: top_100_liked_algos_and_data_structures/levenshtein_distance/levenshtein.c
#include <stdio.h>
#include <stdlib.h>

int levenshtein(char x[], int m, char y[], int n) {
    int cost = 0;
    for (int i = 0; i < n; i ++) {
        if (i == n - 1) {
            cost += abs(m - n);
            return cost;
        }
        if ((x[i] != y[i])) {
            cost += 1;
        }
    }
    return cost;
}

void main() {
    char x[1024] = "faljkdkjdafhhiusahdjfkadhfkjhadjhfiuahsduifjakljhakdhfkjhajwioendkajsdnjlfiwheadfhakjdfhakjhhajdhfjkahdkfhakdfhakjhdfhakjdfahdf";
    char y[1024] = "kafkjahdfkjhakjdshfksdfkjhasdkjfskahfkjsadkfjhaskdhfhajkfvakhdghasdfauhuiawrhbduabudfajfdnaiuwedhbfnwadiuhbfnkaergfuhbjnadsfhvbhjwadkfjvj";
    int m = sizeof(x) / sizeof(x[0]);
    int n = sizeof(y) / sizeof(y[0]);
    printf("%d\n", levenshtein(y, n, x, m));
}

File: top_100_liked_algos_and_data_structures/levenshtein_distance/levenshtein.rs
fn levenshtein(x: String, y: String) -> i32 {
    let (m, n, mut cost) = (x.len(), y.len(), 0);
    if m == 0 {return n as i32}
    if n == 0 {return m as i32}

    for i in 0..n {
        if i == n-1 {
            cost += ((m - n) as i32).abs();
            return cost;
        }
        if x.chars().nth(i).unwrap() != y.chars().nth(i).unwrap() {cost += 1}
    }
    return cost;

}

fn main() {

    let y = String::from("kafkjahdfkjhakjdshfksdfkjhasdkjfskahfkjsadkfjhaskdhfhajkfvakhdghasdfauhuiawrhbduabudfajfdnaiuwedhbfnwadiuhbfnkaergfuhbjnadsfhvbhjwadkfjvj");
    let x = String::from("faljkdkjdafhhiusahdjfkadhfkjhadjhfiuahsduifjakljhakdhfkjhajwioendkajsdnjlfiwheadfhakjdfhakjhhajdhfjkahdkfhakdfhakjhdfhakjdfahdf");

    println!("{}", levenshtein(y, x));
}